auth/src/main/java/org/aerogear/auth/credentials/IntegrityCheckParameters.java

```diff
@@ -0,0 +1,69 @@
+package org.aerogear.auth.credentials;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+public class IntegrityCheckParameters implements IIntegrityCheckParameters {
+
+    private String issuer;
+    private String audience;
+    private String publicKey;
+
+    public IntegrityCheckParameters(final String audience, final String issuer, final String publicKey) {
+        this.issuer = issuer;
+        this.audience = audience;
+        this.publicKey = publicKey;
+    }
+
+    public IntegrityCheckParameters() {}
+
+    /**
+     * Get the audience parameter.
+     * @return audience parameter.
+     */
+    public String getIssuer() {
+        return this.issuer;
+    }
+
+    /**
+     * Get the issuer parameter.
+     * @return issuer parameter.
+     */
+    public String getAudience() {
+        return this.audience;
+    }
+
+    /**
+     * Get the PEM encoded public key (RSA).
+     * @return public key parameter.
+     */
+    public String getPublicKey() { return this.publicKey; }
+
+    public String serialize() throws JSONException {
+        return new JSONObject()
+            .put("audience", this.audience)
+            .put("issuer", this.issuer)
+            .put("publicKey", this.publicKey)
+            .toString();
+    }
+
+    /**
+     * Return json representation of the parameters
+     * @return json string representation of parameters
+     */
+    public static IntegrityCheckParameters deserialize(final String serializedParams) throws JSONException {
```

Comment:
Kill the exception.

If you have to throw a checked exception (pro tip, you don't), make it more specific to the task at hand.  For instance if there is no "issuer" property on the object then throwing as JSON exception is less useful than throwing a new IllegalArgumentException with a useful message.

Also, add that the object needs to be a json object with those parameters in the header.

Also also, if all we are doing is storing this in shared preferences can we make it a parcelable instead?

Reply:
@secondsun I'm not sure I understand how parcelable would help here?